ansible_collections/arista/cvp/plugins/module_utils/device_tools.py

```diff
@@ -605,6 +675,10 @@ def check_device_exist(self, user_inventory: DeviceInventory, search_mode):
                 if self.is_device_exist(device.system_mac) is False:
                     device_not_present.append(device.system_mac)
                     MODULE_LOGGER.error('Device not present in CVP but in the user_inventory: %s', device.system_mac)
+            elif search_mode == FIELD_SERIAL:
```

Comment:
I had to change this to
```
            elif self.__search_by == FIELD_SERIAL or search_mode == FIELD_SERIAL:
                if self.is_device_exist(device.serial_number, search_mode=FIELD_SERIAL) is False:
                    device_not_present.append(device.serial_number)
                    MODULE_LOGGER.error('Device not present in CVP but in the user_inventory: %s', device.serial_number)
```
otherwise `search_mode` was getting set to `hostname`always

Reply:
also device_tools.py has to be adjusted to handle non-existing configlet or parent container case correctly

Reply:
> I had to change this to
> 
> ```
>             elif self.__search_by == FIELD_SERIAL or search_mode == FIELD_SERIAL:
>                 if self.is_device_exist(device.serial_number, search_mode=FIELD_SERIAL) is False:
>                     device_not_present.append(device.serial_number)
>                     MODULE_LOGGER.error('Device not present in CVP but in the user_inventory: %s', device.serial_number)
> ```
> 
> otherwise `search_mode` was getting set to `hostname`always

I would go with this at the top of `manager` function as it is entrypoint

```python
def manager(self, user_inventory: DeviceInventory, search_mode: str = FIELD_HOSTNAME, apply_mode: str = 'loose'):
    response = CvAnsibleResponse()
    self.__search_by = search_mode
```

So we will have consistent behavior with no reason to complexify with `OR`

Reply:
> also device_tools.py has to be adjusted to handle non-existing configlet or parent container case correctly

Would be better to track this change in a specific issue.